contracts/core/Resolving/ResolveBlocker.sol

```diff
@@ -0,0 +1,133 @@
+    /// @notice The interface for the sports AMM manager
+    ISportsAMMV2Manager public manager;
+
+    /// @notice Mapping to track if a game is blocked for resolution
+    /// @dev Maps gameId to a boolean indicating if the game is blocked
+    mapping(bytes32 => bool) public gameIdBlockedForResolution;
+
+    /// @notice Mapping to track if a game has been unblocked by an admin
+    /// @dev Maps gameId to a boolean indicating if the game was unblocked by an admin
+    mapping(bytes32 => bool) public gameIdUnblockedByAdmin;
+
+    /* ========== CONSTRUCTOR ========== */
+
+    /// @notice Initializes the contract with the owner, SportsAMMV2Data, and manager addresses
+    /// @param _owner The address of the contract owner
+    /// @param _sportsAMMV2Data The address of the SportsAMMV2Data contract
+    /// @param _manager The address of the manager contract
+    function initialize(address _owner, address _sportsAMMV2Data, address _manager) external initializer {
+        setOwner(_owner);
+        initNonReentrant();
+        sportsAMMData = ISportsAMMV2Data(_sportsAMMV2Data);
+        manager = ISportsAMMV2Manager(_manager);
+    }
+
+    /// @notice Retrieves the blocked and unblocked status for a list of game IDs
+    /// @param gameIds An array of game IDs to check
+    /// @return blockedGames An array of booleans indicating if each game is blocked
+    /// @return unblockedByAdmin An array of booleans indicating if each game is unblocked by admin
+    function getGamesBlockedForResolution(
+        bytes32[] memory gameIds
+    ) external view returns (bool[] memory blockedGames, bool[] memory unblockedByAdmin) {
+        blockedGames = new bool[](gameIds.length);
+        unblockedByAdmin = new bool[](gameIds.length);
+        for (uint i = 0; i < gameIds.length; i++) {
+            blockedGames[i] = gameIdBlockedForResolution[gameIds[i]];
+            unblockedByAdmin[i] = gameIdUnblockedByAdmin[gameIds[i]];
+        }
+    }
+
+    /// @notice Blocks a list of games for resolution
+    /// @param _gameIds An array of game IDs to block
+    /// @param _reason The reason for blocking the games
+    function blockGames(bytes32[] memory _gameIds, string memory _reason) external onlyWhitelistedForBlock(msg.sender) {
+        _blockGames(_gameIds, true);
+        emit GamesBlockedForResolution(_gameIds, _reason);
+    }
+
+    /// @notice Unblocks a list of games for resolution
+    /// @param _gameIds An array of game IDs to unblock
+    function unblockGames(bytes32[] memory _gameIds) external onlyWhitelistedForUnblock(msg.sender) {
+        _blockGames(_gameIds, false);
+        emit GamesUnblockedForResolution(_gameIds);
+    }
+
+    /// @notice Internal function to block or unblock games
+    /// @param _gameIds An array of game IDs to block or unblock
+    /// @param _blockGame A boolean indicating whether to block (true) or unblock (false) the games
+    function _blockGames(bytes32[] memory _gameIds, bool _blockGame) internal {
```

Comment:
I think we should store the reason for blocking on contract. Easiest would be as a string, but can also use enumaration.